component/component.go

```diff
@@ -96,6 +96,13 @@ type Destroyer interface {
 	DestroyFunc() interface{}
 }
 
+// Exec is responsible for starting the exec plugin to allow a deployment
+// plugin to provide it's own exec functionality.
```

Comment:
I can see this comment being confusing to newcomers to the Waypoint SDK world thinking they have to implement this for `exec` to work. I'd perhaps expand on this a bit more and explain that this is only required if they want to customize how `exec` works, otherwise the entrypoint will handle exec automatically.